### DISCUSSION TITLE
Anthropic/ Claude 无法从客户端请求

### DISCUSSION THREAD

**Discussion by BrandonStudio** [lobehub/lobehub#2442] [General Question] [answered]
### 💻 系统环境

Windows

### 📦 部署环境

Vercel / Zeabur / Sealos

### 🌐 浏览器

Chrome

### 🐛 问题描述

同样的配置，从客户端请求就失败，从服务器请求就成功。
![image](https://github.com/lobehub/lobe-chat/assets/55647556/405948d8-7725-4613-a151-a97d1517831e)

### 📝 补充信息

本地调试发现从客户端输出的请求是OPTIONS而不是POST

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
可能被跨域限制了吧？你看下请求？

[a comment by an automated account is not shown]

**Comment by BrandonStudio** [comment]
第一，我附上了 Cloudflare AI Gateway 的截图，说明请求送达并且返回。返回的结果就是错误的，而且请求为空
第二，我提到了本地测试，客户端发出的请求是 OPTIONS，没有正文。
![image](https://github.com/lobehub/lobe-chat/assets/55647556/29b876ab-5673-4b92-9473-46abc4ce138f)

**Comment by BrandonStudio** [comment]
我知道问题是什么了
浏览器请求会先通过OPTIONS预检去找CORS头，但是Anthropic不支持OPTIONS请求

[a comment by an automated account is not shown]

**Comment by arvinxx** [comment]
OPTIONS 就是跨域请求。 https://juejin.cn/post/7021077647417409550

[a comment by an automated account is not shown]

**Comment by BrandonStudio** [comment, marked as answer]
Upstream: https://github.com/anthropics/anthropic-sdk-typescript/issues/410

**Reply by wangdake5111** [reply]
所以暂时还无法解决吗

**Reply by arvinxx** [reply]
@wangdake5111 对。https://github.com/anthropics/anthropic-sdk-typescript/issues/219
